src/components/list/ProductList.js

```diff
@@ -0,0 +1,308 @@
+import React from "react";
+import { useCart } from "../list/context/cart.context";
+import { useProducts } from "../list/context/product.context";
+import { useWishlist } from "../list/context/wishlist.context";
+import "../list/card.css";
+import "../list/slider.css";
+
+const ProductList = () => {
+  const {
+    products,
+    sortBy,
+    fastDelivery,
+    minDays,
+    iMac,
+    iPad,
+    iWatch,
+    fourStars,
+    threeStars,
+    twoStars,
+    oneStar,
+    filtersDispatch,
+  } = useProducts();
+
+  const { cartState, cartDispatch } = useCart();
+
+  const { wishlistState, wishlistDispatch } = useWishlist();
+
+  const sortData = (products, sortBy) => {
+    if (sortBy === "PRICE-HIGH-TO-LOW") {
+      return products.sort((a, b) => b.price - a.price);
+    } else if (sortBy === "PRICE-LOW-TO-HIGH") {
+      return products.sort((a, b) => a.price - b.price);
+    }
+
+    return products;
+  };
```

Comment:
You can extract out these filter methods in a separate file, say : `filterMethods.js`